### PR TITLE
Bug 1428508 - Remove the ThLog Angular service

### PR DESCRIPTION
**Pull request by camd**
This removes this service entirely and doesn't replace it with anything else.  There was one case in ``dashboard.js`` where logging happened as a result of a ``try.. catch`` so I changed that to a ``console.error`` and added a single-line linting exception.

I figure if we ever get to a point where we think overall logging is useful, we can add it at that time. :)

### CODE REVIEW

File: ui/js/controllers/perf/dashboard.js [hunk 1]
```diff
@@ -25,7 +23,8 @@ perf.controller('dashCtrl', [
                 try {
                     $scope[k] = phDashboardValues[$scope.topic][k];
                 } catch (TypeError) {
-                    $log.error('"'+ k + '" option not found in "' + $scope.topic + '" dashboard config');
+                    // eslint-disable-next-line no-console
+                    console.error('"'+ k + '" option not found in "' + $scope.topic + '" dashboard config');
```
Comment (edmorley) [inline ui/js/controllers/perf/dashboard.js]: I know this is pre-existing, but perhaps it's worth switching this to a template literal?

File: ui/js/services/log.js [hunk 2]
```diff
@@ -1,76 +0,0 @@
...
- * a line like:
- *
- *     ThLogConfigProvider.setWhitelist([
- *         'foo'
- *     ]);
- *
- * Note: even though this is called ThLogConfig, when you configure it, you must
- * refer to it as a ``ThLogConfigProvider`` in ``local.conf.js``.
- */
-treeherder.provider('ThLogConfig', function () {
```
Comment (edmorley) [inline ui/js/services/log.js]: This makes the commented out example config in `sample.local.conf.js` redundant. I think we should just delete the whole local config system too (`sample.local.conf.js` and the `index.js` in the same directory as it) - what do you think?

Comment (camd) [reply ui/js/services/log.js]: I totally agree.  :)

File: ui/js/controllers/perf/dashboard.js (hunk 1 above)
Comment (camd) [reply ui/js/controllers/perf/dashboard.js]: ahh right, of course.  Good call...  :)

**Pull request merged**